### ISSUE TITLE
Assertion failed when using 0x notation ... feeding garbage to SMT solver

### ISSUE THREAD

**Issue by s-zanella** [FStarLang/FStar#416]
```
let a:int = 0x00 // works
let b:nat = 0    // works
let c:nat = 0x00 // Unknown assertion failed
```
In general, there are all sort of troubles when using the 0x notation. Why is that?

**Comment by catalin-hritcu** [comment]
One more instance of "garbage in, garbage out" (#248). For `let c:nat = 0x00` feeding the following query to the SMT solver:
```
(assert (not (or label_2734 (>= (BoxInt_proj_0 (BoxInt 0x00))
                                (BoxInt_proj_0 (BoxInt 0))))))
```
which does report an error:
```
(error "line 29668 column 57: unknown constant x00")
```
but which then gets swallowed by F* and never displayed.

Could this be an error in the logical encoding?

**Comment by nikswamy** [comment]
@msprotz : any idea about what might be going on with this notation? You're more familiar than I am with the numeric literal support.

**Comment by msprotz** [comment]
I finally was able to (I think) fix it while implementing support for machine integers all across the board. We previously kept integer literals as constant strings all the way until the SMT solver,  where they were printed "as is". Z3 doesn't seem to understand the hex notation, so I had to implement in 385e9e809eaf35f316491ddd02f532acf03d9aab some routines to normalize everything to decimal, both in F# and OCaml.
